front/app/component-library/components/CollapsibleContainer/index.tsx

```diff
@@ -102,25 +102,23 @@ const CollapsibleContainer = ({
   };
 
   return (
-    <Box display="flex" flexDirection="column" {...rest}>
-      <Box display="flex" alignItems="center">
-        <TitleButton
-          as="button"
-          aria-expanded={isExpanded}
-          aria-controls={`collapsed-section-${uuid}`}
-          id={`collapse-container-title-${uuid}`}
-          className={isExpanded ? 'expanded' : 'collapsed'}
-          onClick={handleChange}
-        >
-          {title}
-          <ChevronIcon name="chevron-right" />
-        </TitleButton>
-      </Box>
+    <Box {...boxProps}>
+      <TitleButton
+        aria-expanded={isExpanded}
+        aria-controls={`collapsed-section-${uuid}`}
+        id={`collapse-container-title-${uuid}`}
+        className={isExpanded ? 'expanded' : 'collapsed'}
+        onClick={handleChange}
+      >
+        {title}
+        <ChevronIcon name="chevron-right" />
+      </TitleButton>
       <Box
         role="region"
         aria-live="polite"
         id={`collapsed-section-${uuid}`}
         aria-labelledby={`collapse-container-title-${uuid}`}
+        hidden={!isExpanded}
```

Comment:
Following W3C's [example](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/examples/accordion/), which also makes it easier to wrap your head around, imo.